### PR TITLE
Allow curl --max-time to be configurable

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:

In nested virutalized environments, the 1 second max-time is too
low. Just bumping up WAIT_FOR_URL_API_SERVER does not work unless
we bump up the max-time too.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: hack/lib/util.sh [hunk 1]
```diff
@@ -32,7 +32,7 @@ kube::util::wait_for_url() {
   local i
   for i in $(seq 1 "$times"); do
     local out
-    if out=$(curl --max-time 1 -gkfs "$url" 2>/dev/null); then
+    if out=$(curl --max-time 5 -gkfs "$url" 2>/dev/null); then
```
Comment (cblecker) [inline hack/lib/util.sh]: Right now, we try this by default 30 times, with 1 second apart. Total time to run before timeout would be 1 minute. Changing this to 5 means that it takes 6 seconds per retry, or 3 minutes to time out completely. Is 2 a more reasonable number here?

Comment (dims) [reply hack/lib/util.sh]: Ack @cblecker i'll ask the folks running the tests which combination works best ...

cc @kiwik @animationzl

Comment (liu-sheng) [reply hack/lib/util.sh]: Hi @cblecker ,

In our testing,

Comment (cblecker) [reply hack/lib/util.sh]: @liu-sheng Does it still work if you lower 5 down to 2? I'd be more comfortable with that, then all the way up to 5.

Comment (liu-sheng) [reply hack/lib/util.sh]: @cblecker I have tried with 2, seems it also has the failure possibility, but I have recognized that I used a low performance VM to testing. would you like to expose an environment variable to make that configurable ?

Comment (dims) [reply hack/lib/util.sh]: @liu-sheng good point, let me do that

File: hack/local-up-cluster.sh [hunk 2]
```diff
@@ -582,7 +583,7 @@ function start_apiserver {
 
     # Wait for kube-apiserver to come up before launching the rest of the components.
     echo "Waiting for apiserver to come up"
-    kube::util::wait_for_url "https://${API_HOST_IP}:${API_SECURE_PORT}/healthz" "apiserver: " 1 ${WAIT_FOR_URL_API_SERVER} \
+    kube::util::wait_for_url "https://${API_HOST_IP}:${API_SECURE_PORT}/healthz" "apiserver: " ${MAX_TIME_FOR_URL_API_SERVER} ${WAIT_FOR_URL_API_SERVER} \
```
Comment (cblecker) [inline hack/local-up-cluster.sh]: That's not what that does. That's the third variable to `kube::util::wait_for_url` which lines up with the wait time between retries.

https://github.com/kubernetes/kubernetes/blob/1b6b2ee790d5dcab1e6852cc40be598b3ccd3fba/hack/lib/util.sh#L21-L25

Comment (dims) [reply hack/local-up-cluster.sh]: fixed. thanks!

**Comment by dims** [comment]
@cblecker this is ready now.

**Comment by dims** [comment]
@deads2k can you please re-lgtm? lost it in a rebase

**Pull request merged**